cannelloni.cpp

```diff
@@ -140,13 +142,15 @@ int main(int argc, char** argv) {
         localPort = strtoul(optarg, NULL, 10);
         break;
       case 'L':
-        strncpy(localIP, optarg, INET_ADDRSTRLEN);
+        strncpy(localIP, optarg, INET_ADDRSTRLEN-1);
+        localIP[INET_ADDRSTRLEN-1] = '\0';
```

Comment:
Why is this necessary?

`#define INET_ADDRSTRLEN 16` in */usr/include/netinet/in.h*.

According to the man page:

`The strncpy() function is similar, except that at most n bytes of src are copied. Warning: If there is no null byte among the first n bytes of src, the string placed in dest will not be null-terminated. `

"255.255.255.255" is the maximum length in IPv4, total of 15 characters, 16 with a terminating `\0`.
So `strncpy` should always find a `\0` as long as `optarg` is properly terminated (but I don't know whether this can be guaranteed).

Reply:
I agree regarding max IP string length and null termination of optarg. However user can pass (intentionally or not) any string as an argument longer than INET_ADDRSTRLEN which will lead to lack of \0.

Reply:
Ah...good catch! I like it!